### PR TITLE
Pick closest tags for a few packages and avoid random SHA

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Instead of a random SHA, pick a decent git tag closest to what we have today and use that. For these specific packages, since they are non-semver, we can't really use the version, so we just the SHA corresponding to the git tag 

- https://github.com/elazarl/goproxy/compare/c4fc26588b6e...v1.1 (Move from random to v1.11)
- https://github.com/coreos/pkg/compare/v3...97fdf19511ea (Move from random to v3)
- https://github.com/checkpoint-restore/go-criu/compare/v3.11...bdb7599cd87b (move from random to 3.11)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,7 +27,7 @@ require (
 	github.com/boltdb/bolt v1.3.1 // indirect
 	github.com/caddyserver/caddy v1.0.3
 	github.com/cespare/prettybench v0.0.0-20150116022406-03b8cfe5406c
-	github.com/checkpoint-restore/go-criu v0.0.0-20190109184317-bdb7599cd87b // indirect
+	github.com/checkpoint-restore/go-criu v0.0.0-20181120144056-17b0214f6c48 // indirect (3.11 tag)
```
Comment (liggitt) [inline go.mod]: I'm surprised we could backlevel this without go complaining

File: go.mod [hunk 2]
```diff
@@ -38,7 +38,7 @@ require (
 	github.com/containernetworking/cni v0.7.1
 	github.com/coredns/corefile-migration v1.0.4
 	github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
-	github.com/coreos/pkg v0.0.0-20180108230652-97fdf19511ea
+	github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf // (v3 tag)
```
Comment (liggitt) [inline go.mod]: same comment about backleveling 2 years

**Comment by liggitt** [comment]
looks like comments are only preserved on the replace directives

**Comment by dims** [comment]
@liggitt i had to add a regex in the verify script to ignore the comments, looks good to go.

**Comment by liggitt** [comment]
> @liggitt i had to add a regex in the verify script to ignore the comments, looks good to go.

the verify script just runs `hack/update-vendor.sh` and checks for a diff against what is committed. doesn't needing to ignore those diffs mean that mean rerunning `hack/update-vendor.sh` will drop those comments?

**Comment by dims** [comment]
@liggitt y, looks like we either have to augment update script to re-add the comment or just drop this whole idea of adding a comment.

**Comment by liggitt** [comment]
adding the comment to the `replace` directive (not in the require section) should work

**Comment by dims** [comment]
/retest

@liggitt It does! please approve/lgtm. the failures are flakes

File: go.mod [hunk 3]
```diff
@@ -216,7 +216,8 @@ replace (
 	github.com/cenkalti/backoff => github.com/cenkalti/backoff v2.1.1+incompatible
 	github.com/cespare/prettybench => github.com/cespare/prettybench v0.0.0-20150116022406-03b8cfe5406c
 	github.com/chai2010/gettext-go => github.com/chai2010/gettext-go v0.0.0-20160711120539-c6fed771bfd5
-	github.com/checkpoint-restore/go-criu => github.com/checkpoint-restore/go-criu v0.0.0-20190109184317-bdb7599cd87b
+	// 17b0214f6c48 is the SHA for git tag 3.11
+	github.com/checkpoint-restore/go-criu => github.com/checkpoint-restore/go-criu v0.0.0-20181120144056-17b0214f6c48
```
Comment (liggitt) [inline go.mod]: go mod tools will preserve the comment best if it is on the same line, like this:
```
github.com/checkpoint-restore/go-criu => github.com/checkpoint-restore/go-criu v0.0.0-20181120144056-17b0214f6c48 // 17b0214f6c48 is the SHA for git tag 3.11
```

Comment (dims) [reply go.mod]: Ack. Done!

File: go.mod [hunk 4]
```diff
@@ -233,7 +233,7 @@ replace (
 	github.com/coreos/go-oidc => github.com/coreos/go-oidc v2.1.0+incompatible
 	github.com/coreos/go-semver => github.com/coreos/go-semver v0.3.0
 	github.com/coreos/go-systemd => github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
-	github.com/coreos/pkg => github.com/coreos/pkg v0.0.0-20180108230652-97fdf19511ea
+	github.com/coreos/pkg => github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf // 3ac0863d7acf is the SHA for git tag v3
```
Comment (liggitt) [inline go.mod]: we should use the `v4` tag which matches what we're already pointing to (https://github.com/coreos/pkg/commits/v4)

File: go.mod [hunk 5]
```diff
@@ -248,7 +248,7 @@ replace (
 	github.com/docker/libnetwork => github.com/docker/libnetwork v0.8.0-dev.2.0.20190624125649-f0e46a78ea34
 	github.com/docker/spdystream => github.com/docker/spdystream v0.0.0-20160310174837-449fdfce4d96
 	github.com/dustin/go-humanize => github.com/dustin/go-humanize v1.0.0
-	github.com/elazarl/goproxy => github.com/elazarl/goproxy v0.0.0-20170405201442-c4fc26588b6e
+	github.com/elazarl/goproxy => github.com/elazarl/goproxy v0.0.0-20180725130230-947c36da3153 // 947c36da3153 is the SHA for git tag v1.11
```
Comment (liggitt) [inline go.mod]: ok

File: go.mod [hunk 6]
```diff
@@ -216,7 +216,7 @@ replace (
 	github.com/cenkalti/backoff => github.com/cenkalti/backoff v2.1.1+incompatible
 	github.com/cespare/prettybench => github.com/cespare/prettybench v0.0.0-20150116022406-03b8cfe5406c
 	github.com/chai2010/gettext-go => github.com/chai2010/gettext-go v0.0.0-20160711120539-c6fed771bfd5
-	github.com/checkpoint-restore/go-criu => github.com/checkpoint-restore/go-criu v0.0.0-20190109184317-bdb7599cd87b
+	github.com/checkpoint-restore/go-criu => github.com/checkpoint-restore/go-criu v0.0.0-20181120144056-17b0214f6c48 // 17b0214f6c48 is the SHA for git tag 3.11
```
Comment (liggitt) [inline go.mod]: ok

File: go.mod (hunk 4 above)
Comment (dims) [reply go.mod]: Done!

**Pull request merged**